generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -0,0 +1,220 @@
+
+    size_t const pointCount = f.GetPointsCount();
+    m_roadPointCount += pointCount;
+
+    for (int i = 0; i < pointCount; ++i)
+    {
+      IsNewFeatureEnd(f.GetPoint(i), m_uniqueRoadPoints);
+    }
+
+    // Feature length.
+    size_t featureLengthMeters = 0;
+    if (pointCount != 0)
+      featureLengthMeters =
+          MercatorBounds::DistanceOnEarth(f.GetPoint(0), f.GetPoint(pointCount - 1));
+    IncrementStats(featureLengthMeters, m_featureLength);
+
+    // Feature segment length.
+    if (pointCount != 0)
+    {
+      for (int i = 0; i < pointCount - 1; ++i)
+      {
+        size_t const segmentLengthMeters =
+            MercatorBounds::DistanceOnEarth(f.GetPoint(i), f.GetPoint(i + 1));
+        IncrementStats(segmentLengthMeters, m_segLength);
+      }
+    }
+
+    // Feature altitude difference.
+    generator::SrtmTile::THeight startAltitude = 0, endAltitude = 0;
+    if (pointCount != 0)
+    {
+      startAltitude = m_srtmManager.GetHeight(MercatorBounds::ToLatLon(f.GetPoint(0)));
+      endAltitude = m_srtmManager.GetHeight(MercatorBounds::ToLatLon(f.GetPoint(pointCount - 1)));
+    }
+
+    int16_t const altitudeDiff = endAltitude - startAltitude;
+    IncrementStats(altitudeDiff, m_altitudeDiffs);
+
+    // Wave feature factor. Climb minus altitude difference.
+    int32_t climb = 0;
+    if (pointCount != 0)
+    {
+      for (int i = 0; i < pointCount - 1; ++i)
+      {
+        auto const segAltDiff =
+            (m_srtmManager.GetHeight(MercatorBounds::ToLatLon(f.GetPoint(i + 1))) -
+             m_srtmManager.GetHeight(MercatorBounds::ToLatLon(f.GetPoint(i))));
+        if (altitudeDiff >= 0)
+        {
+          // If the feature goes up generaly calculates how many meters it's necessary
+          // to go up following the feature.
+          if (segAltDiff > 0)
+            climb += segAltDiff;
+        }
+        else
+        {
+          // If the feature goes down generaly calculates how many meters it's necessary
+          // to go down following the feature.
+          if (segAltDiff < 0)
+            climb += segAltDiff;
+        }
+      }
+      // Estimates monotony of the feature. If climb == altitudeDiff it's monotonous.
+      // If not it's wavy. The more abs(climb - altitudeDiff) the more wavy the feature.
+      int32_t const waveFactor = climb - altitudeDiff;
+      IncrementStats(waveFactor, m_featureWave);
+    }
+  }
+};
+}  // namespace
+
+int main(int argc, char ** argv)
+{
+  google::SetUsageMessage("This tool extracts some staticstics about features and its altetude.");
+  google::ParseCommandLineFlags(&argc, &argv, true);
+
+  LOG(LINFO, ("srtm_path =", FLAGS_srtm_path));
+  LOG(LINFO, ("mwm_path =", FLAGS_mwm_path));
+
+  classificator::Load();
+  generator::SrtmTileManager manager(FLAGS_srtm_path);
+
+  Processor doProcess(manager);
+  feature::ForEachFromDat(FLAGS_mwm_path, doProcess);
+
+  cout << endl << "doProcess.m_roadCount = " << doProcess.m_roadCount << endl;
```

Comment:
s/doProcess.m_roadCount/Road count:/g
s/doProcess.m_uniqueRoadPoints.size/Num of unique road points:/g
...